firmware/main.c

```diff
@@ -280,6 +282,84 @@ static void cmd_gpio(BaseSequentialStream *chp, int argc, char *argv[]) {
 	chprintf(chp, "Bad gpio '%s'\r\n", pOpt);
 }
 
+//some variables for the blink thread
+static THD_WORKING_AREA(blinkThreadArea, 128);
+thread_t *blinkThread;
+
+
+static THD_FUNCTION(blinking, arg){
+	int i = (int) arg;
+	//endless loop of blinking lights
+	palSetPadMode(pinPorts[i].gpio, pinPorts[i].pin, PAL_MODE_OUTPUT_PUSHPULL);
+	while (!chThdShouldTerminateX()){
+		palSetPad(pinPorts[i].gpio, pinPorts[i].pin);
+		chThdSleepSeconds(1);
+		palClearPad(pinPorts[i].gpio, pinPorts[i].pin);
+		chThdSleepSeconds(1);
+	}
+	palSetPadMode(pinPorts[i].gpio, pinPorts[i].pin, PAL_MODE_INPUT);
+	return;
+}
+
+
+static void cmd_blink(BaseSequentialStream *chp, int argc, char *argv[]) {
+	//start or stop a thread for the blinking lights
+	int i;
+	char *pOpt = NULL;
+	char *sOpt = NULL;
+
+			//parse arguments
+	for(i = 0; i < argc; i++) {
+		if(strcmp(argv[i], "-s") == 0) {
+			if(++i >= argc) continue;
+#ifdef SHELL_OPTION_PARSING_DEBUG
+			chprintf(chp, "detected -s with val: %s\r\n", argv[i]);
+#endif
+			sOpt = argv[i];}
+
+
+
+			else if(strcmp(argv[i], "-p") == 0) {
+				if(++i >= argc) continue;
+#ifdef SHELL_OPTION_PARSING_DEBUG
+			chprintf(chp, "detected -p with val: %s\r\n", argv[i]);
+#endif
+			pOpt = argv[i];
+	}}
+	//if s is off then pin can be empty
+	if(!sOpt || ((strcmp(sOpt, "off") == 0) && pOpt) || ((strcmp(sOpt, "on") == 0) && !pOpt)){
+		chprintf(chp, "Usage: blink -s [state] [-p <pin>] \r\n"
+									"\twith state:\r\n"
+									"\t\ton | off\r\n"
+									"\twith pin (only if state on):\r\n"
+									"\t\t");
+									//all pins available can be blinked
+		for(i = 0; i < sizeof(pinPorts)/sizeof(pinPorts[0]); i++)
+			if(pinPorts[i].as_gpio)
+				chprintf(chp, "%s | ", pinPorts[i].pinNrString);
+
+		chprintf(chp, "\n");
+		return;
+}
+
+
+	//cancel the thread if state is set to mode
+	if (strcmp(sOpt, "off") == 0) chThdTerminate(blinkThread);
```

Comment:
please put the statement to run if the if-clause is true on its own line.

Reply:
also you need to make sure that blinkThread is valid. see below.

Reply:
there are still a few if-statements on one line. e.g. lines 314 or 321.

Reply:
you can fix indentation using '=' if you are using `vim` as an editor.
just ask vim for the help on that via   :help =